### ISSUE TITLE
Move `VM Image List` off of GitHub

### ISSUE THREAD

**Issue by JasonRShaver** [Azure/azure-cli#4611]
Notice in the cloud.py, we have a hardcoded path to a file hosted in github. It means every time users execute `az vm image list`, the command will fetch the file from raw.githubusercontent.com. This render a few issues:

1. Why do we rely on a third party to host files for runtime execution?
2. Github is not a file share service. It is a source control service. That means the url we relies on is not 100% reliable. 
3. Github can throttle the access to this file just like they throttle the release files, which will break our command. 
4. Github can be blocked in some subnet or behind a firewall, which will make this command unusable.
  5.  Github won’t guarantee the performance of this file host.
 
I understand that the logic behind this is that this fire will be updated after CLI is released so as to keep to the latest image types list. However, we should have this file publish to a storage account under our control to guarantee the user experience.
 
```
endpoints=CloudEndpoints(
  | management='https://management.core.windows.net/',
  | resource_manager='https://management.azure.com/',
  | sql_management='https://management.core.windows.net:8443/',
  | batch_resource_id='https://batch.core.windows.net/',
  | gallery='https://gallery.azure.com/',
  | active_directory='https://login.microsoftonline.com',
  | active_directory_resource_id='https://management.core.windows.net/',
  | active_directory_graph_resource_id='https://graph.windows.net/',
  | active_directory_data_lake_resource_id='https://datalake.azure.net/',
  | vm_image_alias_doc='https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json'),    # pylint: disable=line-too-long
  |  
```

**Comment by qwordy** [comment]
@JasonRShaver I agree. It's better to use Azure Storage for file hosting.
@jseely Have you run `az cloud set --name AzureUSGovernment`? What does "no public internet access" mean? Can you access Azure public cloud resource?

**Comment by jseely** [comment]
This is for a government client. They have their environment locked down. I would assume that would mean they would not have access to Azure Public Cloud resources either. I can try to get more details...

Is it not possible to have the CLI pull the data required from the Resource Provider?

**Comment by yugangw-msft** [comment]
@jseely 
This doc contains the alias for a few commonly used images.  It is co-maintained by CLI and Compute service team, you can bypass the query by providing the urn. here are the examples to pull from the RP
```bash
(env) C:\sdk\azure-cli>az vm image list -otable --publisher canonical --offer ubuntu --sku 18.04 --all --query "[].urn"
Result
------------------------------------------------------
Canonical:UbuntuServer:18.04-DAILY-LTS:18.04.201907180
Canonical:UbuntuServer:18.04-DAILY-LTS:18.04.201908011
Canonical:UbuntuServer:18.04-DAILY-LTS:18.04.201908070
Canonical:UbuntuServer:18.04-DAILY-LTS:18.04.201908131
Canonical:UbuntuServer:18.04-DAILY-LTS:18.04.201908140
Canonical:UbuntuServer:18.04-LTS:18.04.201804262
...

(env) C:\sdk\azure-cli>az vm create --image Canonical:UbuntuServer:18.04-LTS:18.04.201804262 -g myRG -n myVM
```

**Comment by jseely** [comment]
@yugangw-msft in our usage the `--image` value that we provide is a Managed Image. Are you saying it shouldn't be looking up the `githubusercontent.com` content in this case?

**Comment by yugangw-msft** [comment]
No, it shouldn't be.
For managed images which were captured/built by customers, they have 2 kinds of residences:
1. As a standalone images, which can be retrieved through `az imge list`
2. In a shared image gallery which can be retrieved through  `az sig image-version list`

**Comment by qwordy** [comment]
Close issue for it's inactive for a long time.

**Comment by anEXPer** [comment]
This is not inactive for a long time. This is an active issue we still have.

**Comment by qwordy** [comment]
> 
> 
> This is not inactive for a "long time." This is an active issue we still have. Two weeks when the last comment is "yeah it shouldn't be doing this" doesn't seem like a closable issue to me.

OK. I don't think Github is a file server.

**Comment by jseely** [comment]
I have opened this PR #10658 which should fix the issue we were running into. 

I also noticed a remark in https://github.com/Azure/azure-powershell/issues/8027 from @BorisB2015 that a solution for this may be to add the aliases doc, that's currently on githubusercontent.com, into the build process for both the Powershell module and the Azure CLI. I think that may be a better long term solution as the image aliases will then also be made available to uses in Azure Environments without public internet access. (Note: I have two large customers in the Defense and Banking industries where this is the case)

**Comment by qwordy** [comment]
https://github.com/Azure/azure-rest-api-specs/blob/master/arm-compute/quickstart-templates/aliases.json
This is the file that we use to convert alias to urn. 
@jseely Thanks for creating the PR.
I agree. Let the CI task copy the latest file to project as a resource file. Then CLI/Powerver no longer need to visit github.

**Comment by qwordy** [comment]
> 
> 
> https://github.com/Azure/azure-rest-api-specs/blob/master/arm-compute/quickstart-templates/aliases.json
> This is the file that we use to convert alias to urn.
> @jseely Thanks for creating the PR.
> I agree. Let the CI copy the latest file to project as a resource file. Then CLI/Powershell can read from it if there is no internet access.

**Comment by qwordy** [comment]
I merged the PR. It will be available in Azure 2.0.76. Official Release: 11/04/2019.
PR: [Compute] vm create: Add a local copy of image alias file so that it can be accessed in a restricted network environment
Environment like Azure GOV can't access Internet other than servers of Azure. When creating VM, Azure CLI visit a image alias file in Github to translate image alias to full urn. This patch intends to add a local copy of this file as a backup in case of network issue.
